docs/hostconf-token.md

```diff
@@ -0,0 +1,160 @@
+# Host configuration token
+
+**work in progress**
+
+Host configuration tokens are used to proof that a host is eligible to enroll
+into an identity domain. A host requests a token from the *idmscv-backend*,
+then presents it to the enrollment agent on an IPA server. The enrollment
+agent verifies the token and on success, enrolls the host. The process uses
+standard JSON Web Keys (JWKs) and JSON Web Signatures (JWS).
+
+The signing JWKs are generated by the *idmsvc-backend*. Enrollment agents
+download a JWK Set and revoked key ids, whenever they register or update
+domain data. This usually happens once a day.
+
+
+## JSON Web Keys (JWK)
+
+Host configuration tokens are signed with an asymmetric JWK. All JWKs are
+elliptic curve (EC) keys with curve P-256, which is a fast and FIPS approved
+algorithm. The key identifier (kid) is based on the SHA-256 fingerprint
+according [RFC7638](https://datatracker.ietf.org/doc/html/rfc7638),
+URL-safe base64 encoded and truncated to 8 characters. Additionally, the keys
+have an expiration time `exp` attribute, which has the same mean as `exp` in
+[RFC7519](https://datatracker.ietf.org/doc/html/rfc7519) JWT.
+
+* `kid`: 8 characters (SHA-256 fingerprint)
+* `kty`: `EC`
+* `crv`: `P-256`
+* `alg`: `ES256`
+* `use`: `sig`
+* `exp`: expiration time as Unix time stamp integer
+
+Example:
+
+```json
+{
+    "alg": "ES256",
+    "crv": "P-256",
+    "exp": 1704261209,
+    "kid": "7lkFVyKx",
+    "kty": "EC",
+    "use": "sig",
+    "x": "dGFSfEJTinH76FFXus90CVn6r5F_FGThLjWrnmMZ3Os",
+    "y": "p4BOLD0REq9BbKpty0nJxZ95nNFeIrxDHH9S4dMsk7M"
+}
+```
+
+### Encryption at rest
+
+Private JWKs are encrypted with standard AES-GCM AEAD algorithm with a random
+nonce. The nonce is pre-pended to the cipher text. The symmetric encryption
+key is derived from the app secret using HKDF-SHA256.
+
+
+### Database schema
+
+* `id` serial
+* `created_at` timestamp
+* `updated_at` timestamp
+* `deleted_at` timestamp
+* `kid` unique varchar (not NULL)
+* `expiration` timestamp (not NULL)
+* `token` text (not NULL)
+* `encrypted` byte array (NULL-able)
+
+The `token` field contains the serializes JSON string representation of the
+**public** JWK and `encrypted` is the encrypted private JWK. Revoked keys
+have a NULL `encrypted` field.
+
+
+## Signed token (JWS / JWT)
+
+The host configuration token uses
+[RFC 7519](https://datatracker.ietf.org/doc/html/rfc7515) JWS (JSON Web
+Signature) and [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519)
+JWT (JSON Web Token) standards. The tokens are **not** JWTs in compact
+notation. Instead they are JWS with one or multiple issuers in JSON string
+notation. Multiple issuers enable smooth key rotation. Old keys are slowly
+phased out while a new key is introduced.
+
+**Header**
+- `kid`: always set
+- `alg`: `ES256`
+
+**Registered JWT claims**
+- issuer (`iss`) must be `"idmsvc/v1"`
+- subject (`sub`) is RHSM cert subject "CN"
+- audience (`aud`) must be `"register host"`
```

Comment:
Bikeshedding: Could we make this `join host` to avoid terminology overload with the "register deployment" usecase, and align with the feature name (_Domain Join_)?

Reply:
I have changed the claim to `join host` in the doc and ipa-hcc.